### PR TITLE
Move date picker

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is BC.

Part of https://github.com/sonata-project/SonataAdminBundle/issues/7156

## Changelog

```markdown
### Changed
- Use datetime picker library from `sonata-project/form-extensions`.
```

### CODE REVIEW

**Review by VincentLanglet — DISMISSED** [review]

File: assets/js/app.js [hunk 1]
```diff
@@ -21,15 +21,7 @@ import 'jquery.scrollto';
 import 'jquery-ui/ui/widget';
 import 'jquery-ui/ui/widgets/sortable';
 import 'bootstrap';
-import moment from 'moment';
```
Comment (VincentLanglet) [inline assets/js/app.js:24]: We have some code related to moment https://github.com/sonata-project/SonataAdminBundle/blob/18732411a37bd7d8831dcafcbb3f8fd462e8dd7a/src/Twig/CanonicalizeRuntime.php#L22

What should we do about it ?

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: assets/js/app.js (hunk 1 above)
Comment (jordisala1991) [reply assets/js/app.js:24]: The ideal would be to move it to form extensions too, and deprecate it here. When upgrading the date picker library it wont be a dependency anymore.

We can move the function to form extensions and:

1. Use here the new function
2. On next major of form extensions the function will always return null (and will be deprecated to remove on 3.0)
3. Here we have a logic to not do anything if it returns null. When compat for form-extensions 1.x ends we remove the calls.

**Comment by jordisala1991** [comment]
Requires: https://github.com/sonata-project/form-extensions/pull/338

File: composer.json [hunk 2]
```diff
@@ -36,7 +36,7 @@
         "sonata-project/block-bundle": "^4.11",
         "sonata-project/doctrine-extensions": "^1.8",
         "sonata-project/exporter": "^2.11",
-        "sonata-project/form-extensions": "^1.7.1",
+        "sonata-project/form-extensions": "^1.15@dev",
```
Comment (jordisala1991) [inline composer.json]:
```suggestion
        "sonata-project/form-extensions": "^1.15",
```

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**